src/js/media/html5.js

```diff
@@ -61,6 +61,23 @@ vjs.Html5 = vjs.MediaTechController.extend({
 
     if (this['featuresNativeTextTracks']) {
       this.on('loadstart', vjs.bind(this, this.hideCaptions));
+
+      // native captions load on demand. Sometimes, some of the tracks get loaded in by default
+      // by the user agent. We don't want the user agent to load stuff by default
```

Comment:
Nope, setting the mode to anything before the track has finished loading does nothing.